crates/router/src/connector/dlocal/transformers.rs

```diff
@@ -471,9 +468,13 @@ impl<F> TryFrom<&DlocalRouterData<&types::RefundsRouterData<F>>> for DlocalRefun
     fn try_from(
         item: &DlocalRouterData<&types::RefundsRouterData<F>>,
     ) -> Result<Self, Self::Error> {
-        let amount_to_refund = item.router_data.request.refund_amount.to_string();
         Ok(Self {
-            amount: amount_to_refund,
+            amount: StringMinorUnitForConnector::convert(
+                &StringMinorUnitForConnector,
+                item.amount,
+                item.router_data.request.currency,
+            )
+            .change_context(errors::ConnectorError::AmountConversionFailed)?,
             payment_id: item.router_data.request.connector_transaction_id.clone(),
```

Comment:
can you please do all the conversion in the flow file of connector and not in transformer file

Reply:
for that we have to create another router data but that is not required

Reply:
have new router data for refund, that should not create a problem!